platform/security/core/security-core-services/src/main/java/ddf/security/audit/impl/SecurityLoggerImpl.java

```diff
@@ -81,47 +85,70 @@ private String getUser(Subject subject) {
     return NO_USER;
   }
 
-  private void requestIpAndPortAndUserMessage(Message message, StringBuilder messageBuilder) {
-    requestIpAndPortAndUserMessage(null, message, messageBuilder);
+  private void appendAdditionalAttributes(Message message, StringBuilder messageBuilder) {
+    appendAdditionalAttributes(null, message, messageBuilder);
   }
 
-  private void requestIpAndPortAndUserMessage(
+  private void appendAdditionalAttributes(
       Subject subject, Message message, StringBuilder messageBuilder) {
 
-    String traceId = ThreadContextUtils.getTraceIdFromContext();
+    String traceId = ThreadContextProperties.getTraceId();
     if (StringUtils.isNotEmpty(traceId)) {
-      messageBuilder.append(TRACE_ID).append(" ").append(traceId).append(" ");
+      messageBuilder.append(TRACE_ID).append(" ").append(traceId).append(", ");
     } else {
-      messageBuilder.append(TRACE_ID).append(" ").append("none").append(" ");
+      messageBuilder.append(TRACE_ID).append(" ").append("none").append(", ");
     }
 
     String user = getUser(subject);
-    messageBuilder.append(SUBJECT).append(user);
+    messageBuilder.append(SUBJECT).append(user).append(", ");
     appendConditionalAttributes(subject, messageBuilder);
 
     if (message == null) {
-      messageBuilder.append(" ");
+      addIpAndPort(
+          ThreadContextProperties.getRemoteAddress(),
+          ThreadContextProperties.getRemotePort(),
+          messageBuilder);
     } else {
       HttpServletRequest servletRequest =
           (HttpServletRequest) message.get(AbstractHTTPDestination.HTTP_REQUEST);
       // pull out the ip and port of the incoming connection so we know
       // who is trying to get access
       if (servletRequest != null) {
-        messageBuilder
-            .append(" Request IP: ")
-            .append(servletRequest.getRemoteAddr())
-            .append(", Port: ")
-            .append(servletRequest.getRemotePort())
-            .append(" ");
+        addIpAndPort(
+            servletRequest.getRemoteAddr(),
+            Integer.toString(servletRequest.getRemotePort()),
+            messageBuilder);
       } else if (MessageUtils.isOutbound(message)) {
         messageBuilder
             .append(" Outbound endpoint: ")
             .append(message.get(Message.ENDPOINT_ADDRESS))
-            .append(" ");
+            .append(", ");
       }
     }
   }
 
+  private void addIpAndPort(String remoteAddress, String remotePort, StringBuilder messageBuilder) {
+    if (remoteAddress != null && remotePort != null) {
+      try {
+        // ServletRequest#getRemoteAddr() can return a long loopback IP6 address which is less than
+        // ideal for audit logging
+        remoteAddress =
```

Comment:
During my testing, the remote address returned from the servlet, if local, would be returned as "0:0:0:0:0:0:0:1".